templates/crossplane/pkg/crd_manager.go.tpl

```diff
@@ -4,151 +4,138 @@ package {{ .CRD.Names.Snake }}
-
-// resourceManager is responsible for providing a consistent way to perform
-// CRUD operations in a backend AWS service API for Book custom resources.
-type resourceManager struct {
-	// rr is the AWSResourceReconciler which can be used for various utility
-	// functions such as querying for Secret values given a SecretReference
-	rr acktypes.AWSResourceReconciler
-	// awsAccountID is the AWS account identifier that contains the resources
-	// managed by this resource manager
-	awsAccountID ackv1alpha1.AWSAccountID
-	// The AWS Region that this resource manager targets
-	awsRegion ackv1alpha1.AWSRegion
-	// sess is the AWS SDK Session object used to communicate with the backend
-	// AWS service API
-	sess *session.Session
-	// sdk is a pointer to the AWS service API interface exposed by the
-	// aws-sdk-go/services/{alias}/{alias}iface package.
-	sdkapi svcsdkapi.{{ .SDKAPIInterfaceTypeName }}API
+// Setup{{ .CRD.Names.Camel }} adds a controller that reconciles {{ .CRD.Names.Camel }}.
+func Setup{{ .CRD.Names.Camel }}(mgr ctrl.Manager, l logging.Logger) error {
+	name := managed.ControllerName({{ .APIVersion}}.{{ .CRD.Names.Camel }}GroupKind)
+	return ctrl.NewControllerManagedBy(mgr).
+		Named(name).
+		For(&{{ .APIVersion}}.{{ .CRD.Names.Camel }}{}).
+		Complete(managed.NewReconciler(mgr,
+			cpresource.ManagedKind({{ .APIVersion}}.{{ .CRD.Names.Camel }}GroupVersionKind),
+			managed.WithExternalConnecter(&connector{kube: mgr.GetClient()}),
+			managed.WithReferenceResolver(managed.NewAPISimpleReferenceResolver(mgr.GetClient())),
+			managed.WithConnectionPublishers(),
+			managed.WithLogger(l.WithValues("controller", name)),
+			managed.WithRecorder(event.NewAPIRecorder(mgr.GetEventRecorderFor(name)))))
 }
 
-// concreteResource returns a pointer to a resource from the supplied
-// generic AWSResource interface
-func (rm *resourceManager) concreteResource(
-	res acktypes.AWSResource,
-) *resource {
-	// cast the generic interface into a pointer type specific to the concrete
-	// implementing resource type managed by this resource manager
-	return res.(*resource)
+type connector struct {
+	kube client.Client
 }
 
-// ReadOne returns the currently-observed state of the supplied AWSResource in
-// the backend AWS service API.
-func (rm *resourceManager) ReadOne(
-	ctx context.Context,
-	res acktypes.AWSResource,
-) (acktypes.AWSResource, error) {
-	r := rm.concreteResource(res)
-	if r.ko == nil {
-		// Should never happen... if it does, it's buggy code.
-		panic("resource manager's ReadOne() method received resource with nil CR object")
+func (c *connector) Connect(ctx context.Context, mg cpresource.Managed) (managed.ExternalClient, error) {
+	cr, ok := mg.(*{{ .APIVersion}}.{{ .CRD.Names.Camel }})
+	if !ok {
+		return nil, errors.New(errUnexpectedObject)
 	}
-	observed, err := rm.sdkFind(ctx, r)
+	cfg, err := awsclient.GetConfig(ctx, c.kube, mg, cr.Spec.ForProvider.Region)
```

Comment:
In AWS Provider, we actually treat `region` as one of the spec fields and it doesn't really appear in any `*Input` queries but in `*aws.config` object. It'd be great to have support for additional fields in the generation code. Another option could be a `CustomFields` struct that's `inline`d and filled in `custom.go` file that's manually written.